### PR TITLE
initial commit of max weight bipartite matching

### PR DESCRIPTION
**Pull request by gahokas**

### CODE REVIEW

File: jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+ *
+ * Changes
+ * -------
+ * 30-Sep-2015 : Initial revision (GA);
+ *
+ */
+
+
+
+/* This class is finds a maximum weight matching of a simple undirected
```
Comment (jsichi) [inline jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java]: Move this to be a Javadoc comment (i.e. starting with /**) right before the class definition.  Also typo:  "This class is finds".

File: jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+	public MaximumWeightBipartiteMatching(final WeightedGraph<V,E> graph, Set<V> vertexPartition1, Set<V> vertexPartition2) {
+		this.graph = graph;
+		partition1 = vertexPartition1;
+		partition2 = vertexPartition2;
+		
+		vertexWeights = new HashMap<V, Long>();
+		hasVertexBeenProcessed = new HashMap<V, Boolean>();
+		isEdgeMatched = new HashMap<E, Boolean>();
+		
+		initializeVerticiesAndEdges();
```
Comment (jsichi) [inline jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java]: Let's use the standard spelling (vertices) uniformly.

File: jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+	
+	private void setVertexWeight(V vertex, Long weight) {
+		vertexWeights.put(vertex, weight);
+	}
+	
+	private long reducedWeight(E edge) {
+		return (long) (vertexWeight(graph.getEdgeSource(edge)) + vertexWeight(graph.getEdgeTarget(edge)) - graph.getEdgeWeight(edge));
+	}
+	
+	private V adjacentVertex(E edge, V vertex) {
```
Comment (jsichi) [inline jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java]: You can use the existing utility method Graphs.getOppositeVertex.

File: jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+		for (V vertex : reachableVerticies.keySet()) {
+			if (isTargetVertex(vertex))
+				continue;
+			for (E edge : graph.edgesOf(vertex)) {
+				if (hasVertexBeenProcessed.get(adjacentVertex(edge, vertex)) && !reachableVerticies.keySet().contains(adjacentVertex(edge, vertex)) && reducedWeight(edge) < beta)
+					beta = reducedWeight(edge);
+			}
+		}
+		
+		if (alpha == 0 || beta == 0)
```
Comment (jsichi) [inline jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java]: This should be an assert instead.

File: jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+			else
+				vertexWeights.put(vertex, vertexWeights.get(vertex) + minValue);
+		}
+		
+	}
+	
+	private Map<V, List<E>> verticesReachableByTightAlternatingEdgesFromVertex(V vertex) {
+		Map<V, List<E>> pathsToVertices = new HashMap<V, List<E>>();
+		pathsToVertices.put(vertex, new ArrayList<E>());
+		_verticesReachableByTightAlternatingEdgesFromVertex(Arrays.asList(vertex), false, pathsToVertices);
```
Comment (jsichi) [inline jgrapht-core/src/main/java/org/jgrapht/alg/MaximumWeightBipartiteMatching.java]: Remove underscore from method name.

**Comment by jsichi** [comment]
Looks good.  I've added a few minor comments on cosmetic stuff; once those are resolved, this is ready to merge.

**Comment by gahokas** [comment]
Thanks for the feedback. Cleaned up per your suggestions.

**Comment by jsichi** [comment]
Thanks!

**Pull request merged**